state/azure/tablestorage/tablestorage.go

```diff
@@ -227,11 +230,11 @@ func (r *StateStore) deleteRow(req *state.DeleteRequest) error {
 	pk, rk := getPartitionAndRowKey(req.Key)
 	entity := r.table.GetEntityReference(pk, rk)
 
-	var etag string
 	if req.ETag != nil {
-		etag = *req.ETag
+		entity.OdataEtag = *req.ETag
+
+		return entity.Delete(false, nil)
 	}
-	entity.OdataEtag = etag
 
 	return entity.Delete(true, nil)
```

Comment:
For my understanding, is there a reason for forcing the delete? It seems like the bool arg only relates to whether the ETag is ignored for the deletion anyway, so it seems like both returns could be consolidated into just the non-forced delete call. Might be useful to add a comment on its necessity.

Reply:
We discussed this via chat. I've added code comments that hopefully help explain this. This is also documented in the tablestorage documentation.